apps/prairielearn/src/pages/instructorAssessmentStatistics/instructorAssessmentStatistics.html.ts

```diff
@@ -0,0 +1,292 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import * as _ from 'lodash';
+import { z } from 'zod';
+
+import { compiledScriptTag, nodeModulesAssetPath } from '../../lib/assets';
+import { AssessmentInstanceSchema, AssessmentSchema, Assessment } from '../../lib/db-types';
+
+export const DurationStatSchema = z.object({
+  median_formatted: z.string(),
+  min_formatted: z.string(),
+  max_formatted: z.string(),
+  mean_formatted: z.string(),
+  median_minutes: z.number(),
+  min_minutes: z.number(),
+  max_minutes: z.number(),
+  mean_minutes: z.number(),
+  threshold_seconds: AssessmentSchema.shape.duration_stat_threshold_seconds,
+  threshold_labels: AssessmentSchema.shape.duration_stat_threshold_labels,
+  hist: AssessmentSchema.shape.duration_stat_hist,
+});
+export type DurationStat = z.infer<typeof DurationStatSchema>;
+
+export const AssessmentScoreHistogramByDateSchema = z.object({
+  date: z.date(),
+  date_formatted: z.string(),
+  number: z.number(),
+  mean_score_perc: z.number(),
+  histogram: z.array(z.number()),
+});
+type AssessmentScoreHistogramByDate = z.infer<typeof AssessmentScoreHistogramByDateSchema>;
+
+export const UserScoreSchema = z.object({
+  duration_secs: z.number(),
+  score_perc: AssessmentInstanceSchema.shape.score_perc,
+});
+type UserScore = z.infer<typeof UserScoreSchema>;
+
+export function InstructorAssessmentStatistics({
+  resLocals,
+  assessment,
+  durationStat,
+  assessmentScoreHistogramByDate,
+  userScores,
+  scoreStatsCsvFilename,
+  durationStatsCsvFilename,
+  statsByDateCsvFilename,
+}: {
+  resLocals: Record<string, any>;
+  assessment: Assessment;
+  durationStat: DurationStat;
+  assessmentScoreHistogramByDate: AssessmentScoreHistogramByDate[];
+  userScores: UserScore[];
+  scoreStatsCsvFilename: string;
+  durationStatsCsvFilename: string;
+  statsByDateCsvFilename: string;
+}) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head'); %>", {
+          ...resLocals,
+          pageTitle: 'Statistics',
+        })}
+        <script src="${nodeModulesAssetPath('lodash/lodash.min.js')}"></script>
+        <script src="${nodeModulesAssetPath('d3/dist/d3.min.js')}"></script>
```

Comment:
I believe this can be removed now, right?

Reply:
Yes, removed in [26d8ced](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/26d8ced6f4b96a0bdd851b8bcbf406187ce99ad5).

Reply:
I mentioned in a previous thread that lodash is needed, but it's only used server-side, so I believe it can be removed client-side as well.

Reply:
Removed in [8a157e8](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/8a157e833810f66048fcf70e2af27e937f981c58).